### DISCUSSION TITLE
scheduled_charging_mode and set_scheduled_charging

### DISCUSSION THREAD

**Discussion by lovekull76** [alandtse/tesla#483] [General]
Hi all,

I have not installed this module yet due to that I have made my own controller (RPi based Tesla-API-harvester/controller -> MQTT - > HASS) for getting and setting scheduled charging according to hourly energy prices.
I would like to use this module instead but I can't see that it reports "scheduled_charging_mode" (and "scheduled_charging_start_time_minutes", only available when mode is "StartAt") or that it can "set_scheduled_charging" in minutes after midnight. Is all this possible? Is it possible to make custom calls to the API (set and get)?

[EDIT] Installed the module, works great and I can find a binary sensor for "Scheduled charging" and I guess that is the "scheduled_charging_mode" (not plugged in so can't test it). 
Also found the attributes "Scheduled charging time" and "..timestamp", looks promising. Still no way to set set_scheduled_charging afaik. But if it is possible to send custom commands it would be possible to do that way.

[Edit_2] And now I found the custom api call function and can set time for scheduled charging!
I just love this module! Great work!

Best regards, L

**Comment by madsdyd** [comment]
Great!

Could you provide an example of setting the scheduled charging? I am also looking into this, would be very helpfull! :-)

**Comment by InTheDaylight14** [comment]
https://github.com/alandtse/tesla/wiki/Scheduled-Charging-and-Departure

**Reply by madsdyd** [reply]
Thank you so much! I never even considered that it was in the wiki. *douh*.

**Comment by lovekull76** [comment]
Did not see the wiki but I did something very similar in HASS using different input_datetime objects depending on how many % I need to charge up.

```
service: tesla_custom.api
data:
  command: SCHEDULED_CHARGING
  parameters:
    path_vars:
      vehicle_id: "{{ state_attr('binary_sensor.korren_online', 'id') }}"
    enable: true
    time: >
      {% set ns = namespace(t = 0) %} 
      {% set ns.t = (states("number.korren_charge_limit")|int(0) - states("sensor.korren_battery")|int(0))/13.33 %} 
      {% if ns.t < 1.5 %}
        {{states('input_datetime.best_price_for_1h_load')[0:2] | int * 60}}
      {% elif ns.t < 2.5 %}
        {{states('input_datetime.best_price_for_2h_load')[0:2] | int * 60}}
      {% else %}
        {{states('input_datetime.best_price_for_3h_load')[0:2] | int * 60}}
      {% endif %}
    wake_if_asleep: true
```
This script runs when the charger is being plugged in at home and at a specific time at night if already plugged in (at home).

**Comment by MatTurac** [comment]
Hi
I tried to make a script but i can't get it to work, it says 
"Stopped because an error was encountered at 29. december 2023 kl. 11.40.45 (runtime: 63.24 seconds)"

I tried with an without mail, with and without hardcoded vehicle id

any one that can see what i am doing wrong?

![2023-12-29_11-42-50](https://github.com/alandtse/tesla/assets/60450759/bd7fe589-c422-4137-9e35-0525fe8658a8)
![2023-12-29_11-43-27](https://github.com/alandtse/tesla/assets/60450759/5ca65e8c-8f75-49cc-baaa-16bd9f38d476)
![2023-12-29_11-43-46](https://github.com/alandtse/tesla/assets/60450759/66e8a1b3-fbca-496a-a6dd-85e79e0ae64d)
![2023-12-29_11-44-40](https://github.com/alandtse/tesla/assets/60450759/b29a3c5e-74cd-42d7-bd38-1b2bdbe6232a)
